### PR TITLE
Added minimal TransformerBridge support for Qwen2MoeForCausalLM

### LINKED ISSUE #1464: [Proposal] Add Qwen2-MoE shared+routed expert adapter (Qwen2MoeForCausalLM)
### Proposal

Add a TransformerBridge adapter for `Qwen2MoeForCausalLM` (Qwen1.5-MoE-A2.7B, Qwen2-57B-A14B), a mixture-of-experts with **shared (always-on) experts alongside routed experts**.

### Motivation

Qwen2-MoE combines always-on shared experts (4) with finer-grained routed experts (60, 4 active). This routing structure is distinct from the covered Mixtral (pure routed) and Qwen3Moe (which dropped shared experts). It is the strongest *demonstrated* expert-routing interp substrate in the current gap list: multiple 2025-2026 papers do causal expert-importance audits, safety-critical expert identification and masking, and shared-vs-routed knowledge attribution directly on Qwen1.5-MoE. A tiny random checkpoint exists so CI stays cheap; the smallest production model is 14.3B total / 2.7B active (a single high-end GPU, or ~15GB in 4-bit).

Gap scan (2026-06-25): 18 models, ~790K downloads.

### Pitch

Implement the Qwen2-MoE block: shared-expert MLP + sigmoid/softmax-routed expert MLPs with top-k gating, over Qwen2 GQA attention. Most attention plumbing overlaps the existing Qwen2 adapter, and the routed-expert plumbing overlaps Mixtral/Qwen3Moe; the new piece is the shared-expert path. Expose `router_logits` / per-expert outputs as hooks.

- Claude Code users can scaffold with `/add-model-support Qwen/Qwen1.5-MoE-A2.7B`.
- Register at the four sites listed in [contributing.md](../docs/source/content/contributing.md).
- Verify smallest-first: `hyper-accel/ci-random-qwen2-moe-a3b`, then `Qwen/Qwen1.5-MoE-A2.7B` where hardware allows.

### Additional context

- HF docs: https://huggingface.co/docs/transformers/model_doc/qwen2_moe
- Interp usage (causal audit of expert importance, Qwen-MoE): https://arxiv.org/abs/2606.10703
- Checkpoint: https://huggingface.co/Qwen/Qwen1.5-MoE-A2.7B
- Found via the `hf_scraper` architecture-gaps pass (2026-06-25).

### Checklist

- [ ] I have checked that there is no similar [issue](https://github.com/TransformerLensOrg/TransformerLens/issues) in the repo (**required**)

### PR DESCRIPTION
**Pull request by Canonik**
## Description

Added minimal TransformerBridge support for "Qwen2MoeForCausalLM".

This adapter reuses the existing Qwen2 attention path and existing MoE bridge components, while adding the Qwen2-MoE-specific shared expert path.

The PR includes :
- registering/detecting "Qwen2MoeForCausalLM"
- mapping Qwen2-MoE config fields from HF
- Qwen2-style attention/GQA plumbing
- routed expert MoE path
- shared expert MLP path
- shared expert gate
- local tiny config HF parity tests
- cache coverage for MoE related hook points

Fixes #1464

## Type of change

- [x] New model / architecture support
- [x] Tests

## Testing

Ran the narrow Qwen2-MoE checks:

```bash
UV_NO_SYNC=1 uv run pytest tests/unit/model_bridge/supported_architectures/test_qwen2_moe_adapter.py
UV_NO_SYNC=1 uv run pytest tests/integration/model_bridge/test_qwen2_moe_bridge.py
UV_NO_SYNC=1 uv run pytest tests/unit/tools/test_model_registry.py -k TestRegistrySyncedWithFactory

### CODE REVIEW

**Comment by Canonik** [comment]
should be good to go

**Comment by jlarson4** [comment]
Great work on this @Canonik! Thank you for getting this taken care of, I am merging now
